sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala

```diff
@@ -239,22 +239,98 @@ object GpuDivideUtil {
   }
 }
 
+abstract class GpuDecimalDivide(left: Expression, right: Expression) extends GpuDivModLike with
+  Serializable {
+  // Override the output type as a special case for decimal
+  override def dataType: DataType = (left.dataType, right.dataType) match {
+    case (l: DecimalType, r: DecimalType) => GpuDivideUtil.decimalDataType(l, r)
+    case _ => super.dataType
+  }
+
+  def getNewDecimalType(outputScale: Int, l: DecimalType, r: DecimalType): DecimalType = {
+    val newScale = if (outputScale + r.scale > l.scale) {
+      outputScale + r.scale
+    } else {
+      l.scale - outputScale
+    }
+    val newType = DecimalType(Math.min(Decimal.MAX_LONG_DIGITS, l.precision + newScale),
+      Math.min(Decimal.MAX_LONG_DIGITS, newScale))
+    val raise = newScale - l.scale
+    require(raise <= Decimal.MAX_LONG_DIGITS, s"$newType  is not supported for GPU processing yet.")
```

Comment:
Is there ever a case where this will throw? If so then we really need the checks in tagging before we get to the point that this would throw an exception.